SonarQube.Common/AnalysisProperties/EnvScannerPropertiesProvider.cs

```diff
@@ -0,0 +1,65 @@
+﻿using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace SonarQube.Common
+{
+    /// <summary>
+    /// Provides properties from the environment
+    /// </summary>
+    public class EnvScannerPropertiesProvider : IAnalysisPropertyProvider
+    {
+        private const string ENV_VAR_KEY = "SONARQUBE_SCANNER_PARAMS";
+        private readonly IEnumerable<Property> properties;
+
+        public static bool TryCreateProvider(ILogger logger, out IAnalysisPropertyProvider provider)
+        {
+            if (logger == null)
+            {
+                throw new ArgumentNullException("logger");
+            }
+
+            provider = null;
+            try
+            {
+                provider = new EnvScannerPropertiesProvider(Environment.GetEnvironmentVariable(ENV_VAR_KEY));
+                return true;
+            } catch (Exception e)
+            {
+                logger.LogError(Resources.ERROR_FailedParsePropertiesEnvVar, ENV_VAR_KEY);
+            }
+            return false;
+        }
+
+        public EnvScannerPropertiesProvider(string json)
+        {
+            if (json == null)
```

Comment:
I would use ternay here but that's a matter of taste so you can keep it like this if you prefer.